src/terminal/terminal.c

```diff
@@ -1500,6 +1501,87 @@ int guac_terminal_send_string(guac_terminal* term, const char* data) {
 
 }
 
+int guac_terminal_send_clipboard(guac_terminal *term) {
+    char *filtered = guac_mem_alloc(term->clipboard->length + 12);
```

Comment:
It may be completely obvious to you why you need to allocate 12 bytes in addition to the clipboard length, here, but it isn't to me. This is one example of a place where 1) more comments throughout the code might be helpful, and 2) this may be a good place for a constant to replace the literal `12`.